test/test_meta.py

```diff
@@ -801,30 +977,32 @@ class TestMeta(TestCase):
     def test_meta(self, device, dtype, op):
         # run the OpInfo sample inputs, cross-referencing them with the
         # meta implementation and check the results are the same.  All
-        # the heavy lifting happens in MetaCrossRefMode
+        # the heavy lifting happens in MetaCrossRefFunctionMode
         func = op.get_op()
+        samples = op.sample_inputs(device, dtype, requires_grad=False)
+        for sample_input in samples:
+            args = [sample_input.input] + list(sample_input.args)
+            kwargs = sample_input.kwargs
+            with MetaCrossRefFunctionMode.push(self, dtype=dtype, device=device):
+                expected = func(*args, **kwargs)
+                if isinstance(expected, torch.Tensor) and op.supports_out:
+                    func(*args, **kwargs, out=expected)
 
-        def do_test(run_excludes_anyway=False):
-            samples = op.sample_inputs(device, dtype, requires_grad=False)
-            for sample_input in samples:
-                args = [sample_input.input] + list(sample_input.args)
-                kwargs = sample_input.kwargs
-                with push_torch_function_mode(partial(MetaCrossRefMode, self, run_excludes_anyway=run_excludes_anyway)):
-                    expected = func(*args, **kwargs)
-                    if isinstance(expected, torch.Tensor) and op.supports_out:
-                        func(*args, **kwargs, out=expected)
-
-        if func in overload_exclude_set:
-            self.skipTest('permanently excluded')
-        elif func in meta_exclude_set and dtype not in (torch.complex128, torch.complex64):
-            try:
-                do_test(run_excludes_anyway=True)
-            except Exception:
-                pass
-            else:
-                self.fail('expected failure, but succeeded')
-        else:
-            do_test()
+    @unittest.skipIf(TEST_WITH_ASAN, "Skipped under ASAN")
+    @onlyNativeDeviceTypes
+    @skipIfCrossRef
+    @suppress_warnings
+    @ops(op_db)
+    def test_dispatch_meta(self, device, dtype, op):
+        func = op.get_op()
+        samples = op.sample_inputs(device, dtype, requires_grad=False)
+        for sample_input in samples:
+            args = [sample_input.input] + list(sample_input.args)
+            kwargs = sample_input.kwargs
+            with MetaCrossRefDispatchMode.push(self, dtype=dtype, device=device):
+                expected = func(*args, **kwargs)
+                if isinstance(expected, torch.Tensor) and op.supports_out:
+                    func(*args, **kwargs, out=expected)
```

Comment:
Did you want to test backward? Or is that coming later?

Reply:
coming later, i'll have to redo the skips when I do this